proto/controller_scheduler.proto

```diff
@@ -0,0 +1,11 @@
+syntax = "proto3";
+
+import "./messages.proto";
+
+service Scheduler{
+  rpc Schedule(scheduler.WorkLoad) returns (scheduler.ScheduleResponse);
+
+  rpc GetWorkerMetric(scheduler.Empty) returns (scheduler.WorkerMetric);
+
+  rpc GetWorkLoadMetrics(scheduler.Empty) returns (scheduler.WorkLoadMetricResponse);
+}
```

Comment:
As described in the [project slides](https://docs.google.com/presentation/d/1oioLrU6d3AgrwbuaufFBVgn6wBtaaLpnMboRPG1ZhUI/edit?usp=sharing), the scheduler should present a server API to both the controller and the agent.

What you're describing here is the service API between the controller and the scheduler, and this should allow for:
1. The controller to send scheduling requests to the scheduler
2. The scheduler to send nodes and instances status updates to the controller

The first part of the API (scheduling requests) is synchronous and is a simple client -> server RPC. The second part is asynchronous and can be modeled through a gRPC stream. An API definition could look like:

```proto
// The Scheduler service for the Controller
service Controller {
     // A request for scheduling an instance of a workload.
     // Success means the instance is pending.
     rpc ScheduleInstance(Workload) returns (google.protobuf.Empty);

     // Get worker and instances status updates.
     // Returns a stream of Status messages.
     rpc GetStatusUpdates(google.protobuf.Empty) returns (stream Status);
}

message Status {
    oneof status {
          InstanceStatus = 1,
          NodeStatus = 2,
    }
}
```

Here again, the controller is not polling for updates but getting them asynchronously from the scheduler.

Reply:
According to what has been said about the `push model` from node agent, it is coherent to keep the same model between `scheduler` and `controller`. However, we are not sure (a lack of protobuf knowledge maybe) to understand if the scheduler exposes a single API endpoint for `controller` and `node-agent` or two ?

Reply:
It will expose 2 different sets of APIs.